aws_topology/stackstate_checks/aws_topology/resources/elb_classic.py

```diff
@@ -13,78 +42,105 @@ class ELBClassicCollector(RegisteredResourceCollector):
+        return self.client.describe_instance_health(LoadBalancerName=elb_name).get("InstanceStates", [])
+
+    def process_elbs(self, elb_names=[]):
+        if elb_names:  # Only pass in LoadBalancerNames if a specific name is needed, otherwise ask for all
+            paginator = self.collect_elb_page(LoadBalancerNames=elb_names)
+        else:
+            paginator = self.collect_elb_page()
+        for elb_page in paginator or []:
+            # The paginator fetches max 20 items, which is the max items that can be fetched with describe_tags
+            tag_page = self.collect_tag_page([elb.get("LoadBalancerName", "") for elb in elb_page])
+            for data in elb_page:
+                instance_health = self.collect_instance_health(data.get("LoadBalancerName"))
+                tags = []
+                # Match the result from the fetched tags with the specific ELB
+                for tag_result in tag_page:
+                    if tag_result.get("LoadBalancerName") == data.get("LoadBalancerName"):
+                        tags = tag_result.get("Tags", [])
+                self.process_elb(LoadBalancerData(elb=data, tags=tags, instance_health=instance_health))
+
     def process_all(self, filter=None):
-        for elb_data_raw in self.client.describe_load_balancers().get('LoadBalancerDescriptions') or []:
-            elb_data = make_valid_data(elb_data_raw)
-            self.process_loadbalancer(elb_data)
-
-    def process_one_loadbalancer(self, name):
-        for elb_data_raw in self.client.describe_load_balancers(
-            LoadBalancerNames=[name]
-        ).get('LoadBalancerDescriptions') or []:
-            elb_data = make_valid_data(elb_data_raw)
-            self.process_loadbalancer(elb_data)
-
-    def process_loadbalancer(self, elb_data):
-        elb_name = elb_data['LoadBalancerName']
-        instance_id = 'classic_elb_' + elb_name
-        elb_data['URN'] = [
-            create_resource_arn(
-                'elasticloadbalancing',
-                self.location_info.Location.AwsRegion,
-                self.location_info.Location.AwsAccount,
-                'loadbalancer',
-                elb_name
+        if not filter or "loadbalancers" in filter:
+            self.process_elbs()
+
+    def process_one_elb(self, elb_name):
+        self.process_elbs([elb_name])
+
+    @transformation()
+    def process_elb(self, data):
+        elb = LoadBalancer(data.elb, strict=False)
+        elb.validate()
+        output = make_valid_data(data.elb)
+        elb_name = elb.LoadBalancerName
+        elb_arn = self.agent.create_arn(
+            "AWS::ElasticLoadBalancing::LoadBalancer", self.location_info, resource_id=elb_name
+        )
+        output["Name"] = elb_name
+        output["Tags"] = data.tags
+        output["URN"] = [elb_arn]
+        self.emit_component(elb_arn, self.COMPONENT_TYPE, output)
+
+        vpc_id = elb.VPCId
+        self.emit_relation(elb_arn, vpc_id, "uses service", {})
```

Comment:
Should we change id?

Reply:
We use ARNs everywhere else, I found the formatting that was used before to be very strange